src/channel_state.ts

```diff
@@ -603,6 +614,153 @@ export class ChannelState<
+   * @param {UserResponse<UserType>} user
+   */
+  _updateUserMessages = (
+    messages: Array<
+      ReturnType<
+        ChannelState<
+          AttachmentType,
+          ChannelType,
+          CommandType,
+          EventType,
+          MessageType,
+          ReactionType,
+          UserType
+        >['formatMessage']
+      >
+    >,
+    user: UserResponse<UserType>,
+  ) =>
+    messages.map((m) => {
+      if (m.user?.id !== user.id) {
+        return m;
+      }
+
+      return {
+        ...m,
+        user,
+      };
+    });
+
+  /**
+   * Updates the message.user property with updated user object, for messages.
+   *
+   * @param {UserResponse<UserType>} user
+   */
+  updateUserMessages = (user: UserResponse<UserType>) => {
+    this.messages = this._updateUserMessages(this.messages, user);
+
+    for (const parentId in this.threads) {
+      this.threads[parentId] = this._updateUserMessages(this.threads[parentId], user);
+    }
+
+    this.pinnedMessages = this._updateUserMessages(this.pinnedMessages, user);
+  };
+
+  /**
+   * @private
+   *
+   * Marks messages from given user as deleted, and returns updated message list
+   *
+   * @param {Array<ReturnType<ChannelState<AttachmentType, ChannelType, CommandType, EventType, MessageType, ReactionType, UserType>['formatMessage']>>} messages
+   * @param {UserResponse<UserType>} user
+   * @param {boolean} hardDelete
+   */
+  _deleteUserMessages = (
+    messages: Array<
+      ReturnType<
+        ChannelState<
+          AttachmentType,
+          ChannelType,
+          CommandType,
+          EventType,
+          MessageType,
+          ReactionType,
+          UserType
+        >['formatMessage']
+      >
+    >,
+    user: UserResponse<UserType>,
+    hardDelete = false,
+  ) =>
+    messages.map((m) => {
+      if (m.user?.id !== user.id) {
+        return m;
+      }
+
+      if (hardDelete) {
+        /**
+         * In case of hard delete, we need to strip down all text, html,
+         * attachments and all the custom properties on message
+         */
+        return ({
+          cid: m.cid,
+          created_at: m.created_at,
+          deleted_at: user.deleted_at,
+          id: m.id,
+          latest_reactions: [],
+          mentioned_users: [],
+          own_reactions: [],
+          parent_id: m.parent_id,
+          reply_count: m.reply_count,
+          status: m.status,
+          thread_participants: m.thread_participants,
+          type: 'deleted',
+          updated_at: m.updated_at,
+          user: m.user,
+        } as unknown) as ReturnType<
+          ChannelState<
+            AttachmentType,
+            ChannelType,
+            CommandType,
+            EventType,
+            MessageType,
+            ReactionType,
+            UserType
+          >['formatMessage']
+        >;
+      } else {
+        return {
+          ...m,
```

Comment:
what about the `user` attribute?

Reply:
It will be part of `...m`

Reply:
will it retain the old user data?

Reply:
Yeah, its supposed to!! Sounds wierd, but thats how backend behaves right now!